src/pydata_sphinx_theme/__init__.py

```diff
@@ -92,22 +92,32 @@ def update_config(app, env):
 
         # try to read the json file. If it's a url we use request,
         # else we simply read the local file from the source directory
-        # it will raise an error if the file does not exist
+        # display a log warning if the file cannot be reached
+        content = None
         if urlparse(json_url).scheme in ["http", "https"]:
-            content = requests.get(json_url).text
+            try:
+                content = requests.get(json_url).text
+            except ConnectionError:
+                pass
```

Comment:
If the json file is to be deployed on github.io as part of the workflow, but it hasn't yet, then this will still fail, as github.io returns a 404, but the contents is still there which fails the json parse later on:
```
>>> req = requests.get("https://example.github.io")
>>> req.ok
False
>>> req.text[:50]
'<!DOCTYPE html>\n<html>\n  <head>\n    <meta http-equ'
```

If you check `.ok` that might fix this:
```suggestion
            try:
                resp = requests.get(json_url)
            except ConnectionError:
                pass
            else:
                if resp.ok:
                    content = resp.text
```

Reply:
I want to catch it earlier to avoid to deal with empty content, I'll use [raise_for_status](https://requests.readthedocs.io/en/latest/api/#requests.Response.raise_for_status) in the try statement